### PR TITLE
Update krb5.conf to match IT recommendations

### PR DESCRIPTION
**Pull request by mbacovsky**
The krb5.conf got outdated and Pyxis service keeps refusing the connections.

## Related Issues and Dependencies

- [TDI-237](https://issues.redhat.com/browse/TDI-237)

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Update krb5.conf according to IT recommendations.

## Description

The krb5.conf got outdated and Pyxis service keeps refusing the connections.

### CODE REVIEW

File: s2i/config/krb5.conf [hunk 1]
```diff
@@ -1,30 +1,36 @@
...
-    dns_canonicalize_hostname = fallback
+  default_realm = IPA.REDHAT.COM
+  dns_lookup_realm = true
+  dns_lookup_kdc = true
+  rdns = false
+  dns_canonicalize_hostname = false
+  ticket_lifetime = 24h
+  forwardable = true
+  udp_preference_limit = 0
+  default_ccache_name = KEYRING:persistent:%{uid}
```
Comment (sochotnicky) [inline s2i/config/krb5.conf]: I would recommend leaving this to either default or setting it as a file in /tmp.

Using keyring could mean that other pods/containers on the same node can access the krb key. Plus I think in some cases the support might not be there?

Comment (mbacovsky) [reply s2i/config/krb5.conf]: @sochotnicky I removed the ccaching setup, could you check please?

File: s2i/config/krb5.conf [hunk 2]
```diff
@@ -1,30 +1,35 @@
...
+    #mkdir /etc/ipa && curl -o /etc/ipa/ca.crt https://password.corp.redhat.com/ipa.crt
+    IPA.REDHAT.COM = {
+       pkinit_anchors = FILE:/etc/ipa/ca.crt
+       pkinit_pool = FILE:/etc/ipa/ca.crt
+       default_domain = ipa.redhat.com
+       dns_lookup_kdc = true
+       # Trust tickets issued by legacy realm on this host
+       auth_to_local = RULE:[1:$1@$0](.*@REDHAT\.COM)s/@.*//
+       auth_to_local = DEFAULT
+    }
```
Comment (harshad16) [inline s2i/config/krb5.conf]:
```suggestion
    }

```

can we please add an empty end line, to respect the standard

**Review by harshad16 — COMMENTED** [review]
/lgtm

just one change needed.

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

:+1:

**Pull request merged**